tests/test_storage.py

```diff
@@ -5,9 +5,11 @@
 
 import io
 import sys
+import tempfile
```

Comment:
You can't use `tempfile` in tests. Everything should be under the `tmp_path` fixture or else you're littering files when failures occur.